Userland/Libraries/LibJS/Interpreter.cpp

```diff
@@ -130,11 +130,31 @@ ThrowCompletionOr<Value> Interpreter::run(Script& script_record)
     return *result.value();
 }
 
-ThrowCompletionOr<Value> Interpreter::run(SourceTextModule&)
+ThrowCompletionOr<Value> Interpreter::run(SourceTextModule& module)
 {
-    auto* error = InternalError::create(global_object(), "Can't run modules directly yet :^(");
-    vm().throw_exception(global_object(), Value { error });
-    return throw_completion(error);
+    // FIXME: This is not a entry point as defined in the spec, but is convenient.
+    //        To avoid work we use link_and_eval_module however that can already be
+    //        dangerous if the vm loaded other modules.
+    auto& vm = this->vm();
+    VERIFY(!vm.exception());
+
+    VM::InterpreterExecutionScope scope(*this);
+
+    auto evaluated_or_error = vm.link_and_eval_module({}, module);
+    VERIFY(!vm.exception());
+    // This error does not set vm.exception so we set that here for the stuff that needs it
+    if (evaluated_or_error.is_throw_completion()) {
+        auto* error = vm.heap().allocate<Exception>(global_object(), *(evaluated_or_error.throw_completion().value()));
+        vm.set_exception(*error);
+        return evaluated_or_error.throw_completion();
+    }
```

Comment:
Since this API is brand new and not being used anywhere yet, do we actually need to do this? Only checking for a throw completion at the call site should be fine.

Reply:
Although it is not directly visible in the commit this is used by `js` with 
```cpp
result = script_or_module.visit(
  [&](auto& visitor) {
      return interpreter.run(*visitor);
  });
```

And `js` currently breaks if you return an error but don't set exception.
(Although I guess we could just set the exception in `js` I'll try that real quick)

Reply:
Ended up fixing this in `js` which is nicer :+1: